recipes/catch2/2.x.x/conanfile.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class ConanRecipe(ConanFile):
+    name = "catch2"
+    description = "A modern, C++-native, header-only, framework for unit-tests, TDD and BDD"
+    topics = ("conan", "catch2", "header-only", "unit-test", "tdd", "bdd")
+    url = "https://github.com/catchorg/Catch2"
+    homepage = url
+    license = "BSL-1.0"
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    exports = "LICENSE.txt"
+
+    generators = "cmake"
+
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "Catch2-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    _build_subfolder = "build_subfolder"
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_TESTING"] = "OFF"
+        cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
+        cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
+        cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
```

Comment:
So, Catch2 maintainers plan on to move towards the proper CMake targets approach, so I figured it would make sense to call build just to future-proof a bit.

Reply:
maybe you can use only `self.copy("catch2.hpp", dst="include", src=self._source_subfolder)` instead of running cmake. Of course, I don't know if running cmake it will result in a different header file.

Reply:
I think it's better to use the install, to not explode the number of folders for each version in case something changes and to rely on project's maintainers to describe what's required.

Reply:
good observation!